proposals/NNNN-android-notifications.md

```diff
@@ -0,0 +1,228 @@
+
+2. "...the notification shows for seconds and disappears, but this is quite a bad UX since all the users will be affected and the number of users that connect their devices to Bluetooth speakers/earphones is huge. We already received multiple bad reviews about this from our BETA testers. So, please consider fixing this asap..."
+
+Some Links from forum discussions in the past years. Similar to the above comments, the users in the below discussion forums complain about these SDL notifications.
+
+[Forum discussions 1](https://eu.community.samsung.com/t5/Other-Smartphones/S8-keep-getting-a-notification-about-smartdevicelink/td-p/542988)
+
+[Forum discussions 2](https://forums.androidcentral.com/ask-question/890465-how-get-rid-sdl.html)
+
+[Forum discussions 3](https://www.android-hilfe.de/forum/huawei-p20-pro.3297/smartdevicelink-meldung-deaktivieren.885217.html)
+
+
+This proposal describes a way in which app developers can control SDL behavior by using APIs to enable or disable it. It also proposes changes in the Android Router service to mitigate notifications issues.
+
+## Proposed solution
+
+Proposed solution is described below.
+
+### Provide API for app developers to enable/disable SDL in Android proxy
+
+The Android proxy should provide an API to enable-disable SDL. The router service should be enabled by default. This would help many app developers to provide a setting in their app so that users can disable it if they don't have access to an SDL enabled system. This API will not influence the behavior of other SDL apps available on the users' device.
+
+If the SDL router service is already connected to the SDL enabled vehicle, the disable API will not have any effect on the current SDL router service session. The settings will take effect in the next run. This avoids app disconnections if the user is connected to an SDL enabled system.
+
+The below example shows a static function that can be used by app developers to enable or disable the router service. The implementation within the SDL broadcast receiver to avoid starting the router service is internal, hence it should be considered a development detail.
+
+```java
+public class SdlRouterService extends Service {
+    private static boolean isRouterServiceEnabled = true;
+    
+    public static void setRouterServiceEnabled(boolean flag) {
+        SdlRouterService.isRouterServiceEnabled = flag;
+    }
+}
+```
+
+### Allow users to Stop SDL foreground services
+
+As seen in user reviews on the play store, many users want a way to stop foreground service notifications. The Android router service notification will provide buttons on foreground notifications, which will stop the router service and hence remove the notifications. All SDL apps connected through this Android Router service will receive `SdlManagerListener.onDestroy` when the user stops Android router service and hence the apps can be notified to stop their own foreground services started for SDL. 
```

Comment:
```suggestion
As seen in user reviews on the play store, many users want a way to stop foreground service notifications. The Android Router Service notification will provide buttons on foreground notifications, which will stop the router service and hence remove the notifications. All SDL apps connected through this Android Router Service will receive `SdlManagerListener.onDestroy` when the user stops Android Router Service and hence the apps can be notified to stop their own foreground services started for SDL. 
```
Please update all references to "Android Router Service" to be capitalized, for consistency and readability.  Thanks!